### PR TITLE
Provide custom defaults and our own build script

### PR DESCRIPTION
**Pull request by johvet**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,6 +17,7 @@ ENV FIP "127.0.0.2"
 ENV FPRT 8125
 ENV FORWARD "False"
 ENV TAGS ""
+ENV ELEMENT_TYPE "SERVER"
```
Comment (ziggythehamster) [inline Dockerfile]: sort

**Review by ziggythehamster — CHANGES_REQUESTED** [review]
You also need to change the default agent conf in here so that all the "main" collectors are enabled = False.

Alternatively you could add a `sed` that does that since it's simple. Perhaps hide that behind an env like `DISABLE_ALL_MAIN_COLLECTORS`

File: Dockerfile (hunk 1 above)
Comment (johvet) [reply Dockerfile]: ok, tried to keep diff small - but fine

File: docker-deploy.sh [hunk 2]
```diff
@@ -1,7 +1,16 @@
 #!/bin/sh
 
+error() {
+  echo $1
+  exit 1
+}
+
+test -z "${VERSION}"         && error "VERSION is required"
+test -z "${DOCKER_USERNAME}" && error "DOCKER_USERNAME is required"
+test -z "${DOCKER_PASSWORD}" && error "DOCKER_PASSWORD is required"
```
Comment (ziggythehamster) [inline docker-deploy.sh]: You don't need this because `docker login` should also not be in here

**Review by ziggythehamster — CHANGES_REQUESTED** [review]

File: docker-deploy.sh (hunk 2 above)
Comment (johvet) [reply docker-deploy.sh]: they had it, so I can remove it if you prefer

**Review by ziggythehamster — APPROVED** [review]

**Pull request merged**